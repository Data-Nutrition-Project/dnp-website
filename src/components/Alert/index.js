import React from "react"

import Accordion from 'react-bootstrap/Accordion'
import Card from 'react-bootstrap/Card'
import classNames from 'classnames'
import PropTypes from "prop-types"
import ReactMarkdown from 'react-markdown'

import styles from "./styles.module.css"

const Alert = props => {
    const severityMap = {
        fyi: '',
        low: 'Low Severity',
        medium: 'Moderate Severity',
        high: 'High Severity'
    }

    return (
        <Accordion defaultActiveKey="0">
            <Card className={styles.card}>
                <Card.Header className={styles.columns}>
                    <div className={classNames(styles.alertColumn, styles.circle, styles[props.severity])}></div>
                    <div className={classNames(styles.alertColumn)}>
                        <p className={classNames(styles.label, styles.titleText)}>Title</p>
                        <p className={classNames(styles.label, styles.subtitleText)}>Type</p>
                    </div>
                    <div className={classNames(styles.alertColumn, styles.spacerColumn)}>
                        <p className={styles.titleText}>{props.title}</p>
                        <p className={styles.subtitleText}>{props.type}</p>
                    </div>
                    <div className={classNames(styles.alertColumn, styles.rightColumn)}>
                        <p className={classNames(styles.bold, styles.subtitleText, styles[props.severity])}>
                            {severityMap[props.severity]}
                        </p>
                        <div className={styles.tagsRow}>
                            <p className={classNames(styles.subtitleText, styles.label)}>Who's impacted</p>
                            {props.tags.map(tag => {
                                return (
                                    <p className={classNames(styles.subtitleText, styles.tag)}>{tag}</p>
                                )
                            })}
                        </div>
                    </div>
                    <Accordion.Toggle className={styles.caret} eventKey="0">
                        <img
                            className={styles.caretImg}
                            src={require("../../images/caret.png")}
                            alt="dropdown toggle"
                        />
                    </Accordion.Toggle>
                </Card.Header>
                <Accordion.Collapse eventKey="0">
                    <div className={styles.content}>
                        <ReactMarkdown source={props.content} />
                    </div>
                </Accordion.Collapse>
            </Card>
        </Accordion>
    )
}

Alert.propTypes = {
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    contentType: PropTypes.oneOf(['text', 'markdown', 'vega-lite']).isRequired,
    severity: PropTypes.oneOf(['fyi', 'low', 'medium', 'high']).isRequired,
    type: PropTypes.oneOf(['completeness', 'accuracy', 'timeliness', 'consistenty', 'accessibility']).isRequired,
    tags: PropTypes.arrayOf(PropTypes.string)
}

Alert.defaultProps = {
    title: 'OH NOESSSS!!!!',
    content: '### Info ###\n\nAHHHHHHHHH!!!!',
    contentType: 'markdown',
    severity: 'high',
    type: 'completeness',
    tags: ['age']
}

export default Alert
