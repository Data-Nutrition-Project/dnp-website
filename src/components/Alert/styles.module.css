@value layout: "../layout.css";

@value alertRed from layout;
@value alertOrange from layout;
@value alertYellow from layout;
@value gray from layout;

.card {
    padding: .5rem;
}

.columns {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    background-color: transparent;
    border: none;
    padding: 0;
}

.alertColumn {
    display: flex;
    flex-direction: column;
    margin-right: .5rem;
}

.alertColumn > p,
.subtitleText {
    margin-bottom: 0;
}

.spacerColumn {
    flex-grow: 2;
}

.rightColumn {
    text-align: right;
}

.circle {
    display: inline-block;
    height: 2rem;
    width: 2rem;
    border-radius: 50%;
    margin-right: 1rem;
}

.low {
    color: alertYellow;
}

.medium {
    color: alertOrange;
}

.high {
    color: alertRed;
}

.circle.low {
    background-color: alertYellow;
}

.circle.medium {
    background-color: alertOrange;
}

.circle.high {
    background-color: alertRed;
}

.titleText {
    font-size: 1rem;
}

.subtitleText {
    font-size: .875rem;
}

.bold {
    font-weight: 700;
}

.label {
    color: gray;
}

.tagsRow {
    display: flex;
    flex-direction: row;
}

.tag {
    margin-left: .5rem;
}

.caret,
.caret:focus {
    border: none !important;
    outline: none;
    background-color: transparent;
    border-color: transparent;
    margin-left: .5rem;
}

.caretImg {
    width: 1rem;
}

.content {
    margin-top: 1rem;
    padding-left: 3rem;
    padding-right: 2.5rem;
}
