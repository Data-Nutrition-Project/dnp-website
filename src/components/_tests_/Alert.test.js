import React from "react"
import renderer from "react-test-renderer"
import Alert from "../Alert"

describe("Alert", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Alert />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
