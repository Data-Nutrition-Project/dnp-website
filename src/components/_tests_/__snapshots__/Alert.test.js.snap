// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  className="accordion"
>
  <div
    className="card card"
  >
    <div
      className="columns card-header"
    >
      <div
        className="alertColumn circle high"
      />
      <div
        className="alertColumn"
      >
        <p
          className="label titleText"
        >
          Title
        </p>
        <p
          className="label subtitleText"
        >
          Type
        </p>
      </div>
      <div
        className="alertColumn spacerColumn"
      >
        <p
          className="titleText"
        >
          OH NOESSSS!!!!
        </p>
        <p
          className="subtitleText"
        >
          completeness
        </p>
      </div>
      <div
        className="alertColumn rightColumn"
      >
        <p
          className="bold subtitleText high"
        >
          High Severity
        </p>
        <div
          className="tagsRow"
        >
          <p
            className="subtitleText label"
          >
            Who's impacted
          </p>
          <p
            className="subtitleText tag"
          >
            age
          </p>
        </div>
      </div>
      <button
        className="caret"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="dropdown toggle"
          className="caretImg"
          src="caret.png"
        />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="collapse show"
    >
      <div
        className="content"
      >
        <h3>
          Info
        </h3>
        <p>
          AHHHHHHHHH!!!!
        </p>
      </div>
    </div>
  </div>
</div>
`;
